htdocs/plugins/preloads/googleanalytics.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * googleanalytics ga4.php
+ * Created by david on 13/02/2022 22:41
+ *
+ */
+
+/**
+ * preload class to include the Google Analytics V4 tag in the Head of the webpage, after replacing the smarty variable with the GA4 property code
+ * This can certainly be improved by checking whether we can, based on the preferences by our users (GDPR and cookies)
+ */
+class IcmsPreloadGoogleanalytics extends icms_preload_Item {
+
+    function eventBeforeFooter()
+    {
+        global $xoopsTpl;
+        global $icmsTheme;
+        global $icmsConfigMetaFooter;
+
+
+        try {
+
+            if ($icmsConfigMetaFooter['use_google_analytics'] && isset($icmsConfigMetaFooter['google_analytics']) && $icmsConfigMetaFooter['google_analytics'] != '') {
+				if(substr( $icmsConfigMetaFooter['use_google_analytics'], 0, 2 ) === "G-")
+				{
+					$this->insertGA4Tag($icmsConfigMetaFooter['use_google_analytics']);
+				}
+				else
+				{
+					$this->insertUniversalAnalyticsTag($icmsConfigMetaFooter['use_google_analytics']);
+				}
+            }
+            else {
+                echo 'error getting icmsConfigMetaFooter';
+            }
+        }
+        catch (Exception $e) {
+            echo $e->getMessage();
+        }
+    }
+
+	function insertUniversalAnalyticsTag(string $UAtag)
+	{
+		$icmsTheme->addScript('', '', '(function(i,s,o,g,r,a,m){i[\'GoogleAnalyticsObject\']=r;i[r]=i[r]||function(){
+  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
+  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
+  })(window,document,\'script\',\'//www.google-analytics.com/analytics.js\',\'ga\');
+
+  ga(\'create\', \'UA-' . $UAtag  . '\', \'auto\');
+  ga(\'send\', \'pageview\');', 'module', '2001');
+
+	}
+
```

Comment:
The current snippet from Google is a bit different and is almost identical to the GA4 snippet, using the Google Tag Manager. Should we switch?

Reply:
It is identical - except for the tag value.

Reply:
I've tested and the new snippet does work with older properties.

Reply:
@skenow any word on this?

Reply:
I'm not sure why I'm not being notified of comments here. Sorry to have missed this.

Having 1 function instead of 2 is cleaner, and when I look at the installation instructions for the UA tag in GA, it shows the same code as it does for the GA-4 tag. 

I think we should change it, so if admins don't switch to GA-4 on their own, when Google automatically does it, things should still work.